superset/mcp_service/flask_singleton.py

```diff
@@ -0,0 +1,78 @@
+"""
+Singleton pattern for Flask app creation in MCP service.
+
+This module ensures that only one Flask app instance is created and reused
+throughout the MCP service lifecycle. This prevents issues with multiple
+app instances and improves performance.
+"""
+
+import logging
+import threading
+from typing import Optional
+
+from flask import Flask
+
+logger = logging.getLogger(__name__)
+
+# Singleton instance storage
+_flask_app: Optional[Flask] = None
+_flask_app_lock = threading.Lock()
+
+
+def get_flask_app() -> Flask:
```

Comment:
Why do we need this?
Under normal circumstances a simple module with just the instance is enough
https://stackoverflow.com/questions/6760685/what-is-the-best-way-of-implementing-a-singleton-in-python

Reply:
ah yea i see it got too complex. let me simplify it.